coralproject/talk
# [CORL-3103]: Make protected email domain bans customizable

## What does this PR do?

These changes update to make protected email domains customizable by tenant in admin configuration --> Moderation --> Users --> Email domain. The default list is the current protected email domains list.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

This adds `protectedEmailDomains` to settings.

## Does this PR introduce any new environment variables or feature flags?

no

## If any indexes were added, were they added to `INDEXES.md`?

n/a

## How do I test this PR?

You can test this PR by going into the admin and updating the protected email domain bans. See that the default is the correct protected email domain list, and you can add and remove domains and save. If you have an incorrectly formatted email domain, you will see a validation error below if you try to save. 

If you try to add a protected email domain that you add to the list to ban all commenter accounts, you will get an error. If you go into the Community tab and try to ban a commenter with an email in the protected email domains list, you will not see an option to do an email domain ban. If you try to ban a commenter with an email not in the protected email domains list, you should see this option.

## Were any tests migrated to React Testing Library?



## How do we deploy this PR?
